plugins/websearch/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+use scout_core::{ Plugin, SearchResult, PluginRegistrar };
+
+mod result;
+use result::{WebSearchResult, SearchEngine };
+
+pub struct WebSearchPlugin {
+	engines: Vec<SearchEngine>
+}
+
+impl WebSearchPlugin {
+	fn new() -> Box<dyn Plugin> {
+		gtk::init().unwrap();
+
+		let engines = vec![SearchEngine {
+			name: "Google".to_string(),
+			base_url: "https://google.com/search".to_string(),
+			query_argument: "q".to_string(),
+			icon: None
+		}];
+
+		Box::new(WebSearchPlugin { engines })
+	}
+}
+
+impl Plugin for WebSearchPlugin {
+	fn get_results(&self, query: &str) -> scout_core::Result<Vec<(usize, Box<dyn SearchResult>)>> {
+		if query.len() == 0 {
+			return Ok(vec![]);
+		}
+
+		Ok(self.engines.iter()
+			.map(|engine| WebSearchResult::new(engine, query))
+			.map(|result| (result.get_ranking(&query), Box::new(result.clone()) as Box<dyn SearchResult>))
```

Comment:
Is this clone necessary?

Reply:
Have you tried without it? I can't see a specific reason why it would be necessary for your code.